src/components/InngestCommHandler.ts

```diff
@@ -438,7 +439,7 @@ export class InngestCommHandler<H extends Handler, TransformedRes> {
         if (stepRes.status === 500 || stepRes.status === 400) {
           return {
             status: stepRes.status,
-            body: stepRes.error || "",
+            body: JSON.stringify(marshalError(stepRes.error)),
```

Comment:
We already marshal in `runStep`, I don't think it's needed twice 🤔

Reply:
We get type issues otherwise.

Reply:
I would guess that if we double-marshaled like this, it would create all errors that looked like `Unknown error: ${JSON.stringify(err)}` Perhaps this is related to the union of the Record and string types in the other comment?

Reply:
Yeah - I was able to validate that in local testing: 
![image](https://user-images.githubusercontent.com/1509457/214173569-6f5115d0-8ac7-4be4-80ee-b0063975310b.png)

Reply:
Yeah, I think this is why the previous stuff returned a string.  I wasn't super into a deep dive into refactoring code I don't own here to be honest — just to unblock this particular issue we're seeing 😬